build-parallel.js

```
@@ -0,0 +1,222 @@
+            console.error('Pre-build steps failed for server build');
+            process.exit(preBuildExitCode);
+        }
+    } catch (error) {
+        console.error('Error running pre-build steps for server build:', error.message);
+        process.exit(1);
+    }
+
+    // Then run parallel server build
+    const pnpmPath = path.join(__dirname, 'node_modules', '.bin', 'pnpm');
+    const workspaceArgs = convertArgsToWorkspaceArgs(argv);
+    const args = [
+        '-r',
+        '--aggregate-output',
+        '--reporter=append-only',
+        '--workspace-concurrency=2',
+        ...workspaceArgs,  // Move workspace filters before 'run'
+        'run',
+        'serverBuild',
+        '--skipPrebuildSteps',
+        '--enableConcurrentTaskBuild'
+    ];
+
+    console.log('Running parallel server build...');
+    console.log('pnpm command:', 'pnpm', args.join(' '));
+    try {
+        const exitCode = await executeCommand(pnpmPath, args);
+        process.exit(exitCode);
+    } catch (error) {
+        console.error('Error running server build:', error.message);
+        process.exit(1);
+    }
+}
+
+// Parse command line arguments
+const command = process.argv[2];
+const additionalArgs = process.argv.slice(3); // Capture all arguments after the command
+
+switch (command) {
+    case 'build':
+        runBuild(additionalArgs);
+        break;
+    case 'serverBuild':
+        runServerBuild(additionalArgs);
+        break;
+    default:
+        console.log('Usage: node build-parallel.js [build|serverBuild] [additional arguments...]');
+        console.log('Commands:');
+        console.log('  build       - Run parallel build');
+        console.log('  serverBuild - Run parallel server build');
+        console.log('');
+        console.log('Examples:');
+        console.log('  node build-parallel.js build --task "MyTask"');
+        console.log('  node build-parallel.js serverBuild --task "@(TaskA|TaskB|TaskC)"');
```

This command silently ignores the wrong taskname instead it should throw an error to avoid typos and inform users of that. 

node build-parallel.js build --task "@(BashV3|NonExistentTask|CmdLineV2)"

---

Fixed this via directory check before running the build operations.